test/e2e/install_test.go

```diff
@@ -99,10 +108,136 @@ var _ = Describe("Operator Install", func() {
+		err = os.WriteFile(filepath.Join(artifactPath, catalogsource.Name+"-catalogsource.yaml"), catalogsourceYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	// Get all Bundles in the namespace and save them to the artifact path.
+	bundles := rukpakv1alpha1.BundleList{}
+	err = c.List(ctx, &bundles, client.InNamespace(""))
+	Expect(err).To(Not(HaveOccurred()))
+	for _, bundle := range bundles.Items {
+		// Save bundle to artifact path
+		bundleYaml, err := yaml.Marshal(bundle)
+		Expect(err).To(Not(HaveOccurred()))
+		err = os.WriteFile(filepath.Join(artifactPath, bundle.Name+"-bundle.yaml"), bundleYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	// Get all BundleDeployments in the namespace and save them to the artifact path.
+	bundleDeployments := rukpakv1alpha1.BundleDeploymentList{}
+	err = c.List(ctx, &bundleDeployments, client.InNamespace(""))
+	Expect(err).To(Not(HaveOccurred()))
+	for _, bundleDeployment := range bundleDeployments.Items {
+		// Save bundleDeployment to artifact path
+		bundleDeploymentYaml, err := yaml.Marshal(bundleDeployment)
+		Expect(err).To(Not(HaveOccurred()))
+		err = os.WriteFile(filepath.Join(artifactPath, bundleDeployment.Name+"-bundleDeployment.yaml"), bundleDeploymentYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	for _, namespace := range namespaces.Items {
+		// let's ignore kube-* namespaces.
+		if strings.Contains(namespace.Name, "kube-") {
+			continue
+		}
+
+		namespaceArtifactPath := filepath.Join(artifactPath, namespace.Name)
+		err = os.Mkdir(namespaceArtifactPath, 0755)
+		Expect(err).To(Not(HaveOccurred()))
+
+		// get all deployments in the namespace and save them to the artifact path.
+		deployments, err := kubeClient.AppsV1().Deployments(namespace.Name).List(ctx, metav1.ListOptions{})
+		Expect(err).To(Not(HaveOccurred()))
+		for _, deployment := range deployments.Items {
+			// Save deployment to artifact path
+			deploymentYaml, err := yaml.Marshal(deployment)
+			Expect(err).To(Not(HaveOccurred()))
+			err = os.WriteFile(filepath.Join(namespaceArtifactPath, deployment.Name+"-deployment.yaml"), deploymentYaml, 0644)
+			Expect(err).To(Not(HaveOccurred()))
+		}
+
+		// Get logs from all pods in all namespaces
+		pods, err := kubeClient.CoreV1().Pods(namespace.Name).List(ctx, metav1.ListOptions{})
+		Expect(err).To(Not(HaveOccurred()))
+		for _, pod := range pods.Items {
+			if pod.Status.Phase != v1.PodRunning {
+				continue
+			}
+			for _, container := range pod.Spec.Containers {
+				logs, err := kubeClient.CoreV1().Pods(namespace.Name).GetLogs(pod.Name, &v1.PodLogOptions{Container: container.Name}).Stream(ctx)
+				Expect(err).To(Not(HaveOccurred()))
+				defer logs.Close()
+
+				// Read logs
+				buf := make([]byte, 1024)
+				for {
+					_, err := logs.Read(buf)
+					if err != nil {
+						break
+					}
+				}
+
+				// Save logs to artifact path
+				err = os.WriteFile(filepath.Join(namespaceArtifactPath, pod.Name+"-"+container.Name+"-logs.txt"), buf, 0644)
```

Comment:
My gut is telling me this will only read 1024 bytes of the log at a time, and only write out the last 1024 bytes? (Although golang surprises me sometimes.)

Reply:
So my understanding is that this will read from the logs 1024 bytes at a time until it encounters an error (ideally and EOF error to signal the end of the file, but the error check doesn't seem to differentiate - and maybe there isn't a need to) and then will break and write the output. I would personally prefer to substitute this for loop with a `buf, err := io.ReadAll(logs)` as it should handle all the reading for us.

Reply:
fixed

Reply:
That's why I previously suggested a small buffer and `io.Copy`...

Reply:
I could be reading it wrong, but it looks like `io.ReadAll()` internally reads a small buffer at a time: https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/io/io.go;drc=dc8e2a6a8ec94f2c98ba20edd57932eba284efb1;l=690

Reply:
Definitely agree with the suggestion of `io.Copy` for writing as the stream is read

Reply:
It starts small, but because this function returns the entire contents, `io.ReadAll` continually increases the size of the buffer as it reads more data.

Reply:
using `io.Copy` now, let me know if that's better